grommet/grommet
# Fix Calendar hover styling

#### What does this PR do?

Updates how `active` calendar date is adjusted when a date is selected / when days grid receives focus. Previously, the firstDayInMonth was incorrectly receiving active treatment upon selection of a date OR if dates were already selection then the old date was receiving active treatment upon selection of a new date.

This approach instead only places the active date on the first day on the month when the Calendar receives focus via keyboard. Otherwise, the active date will always be the most recently selected date.

#### Where should the reviewer start?
src/js/components/Calendar/Calendar.js

#### What testing has been done on this PR?
Tested in Calendar stories.

#### How should this be manually tested?
1. Click on a day on Calendar
   - confirm that hover styling is not on the first day of month
   - confirm that if you now press arrow key that the active date moves to the appropriate adjacent day
2. Tab to calendar days via keyboard
   - confirm that first day in month receives active treatment
   
#### Any background context you want to provide?

#### What are the relevant issues?
Closes #5634 

#### Screenshots (if appropriate)

#### Do the grommet docs need to be updated?
No.

#### Should this PR be mentioned in the release notes?
Yes. Fixed Calendar active day styling.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.
